axelrod/strategies/mindreader.py

```diff
@@ -46,15 +46,25 @@ class MindReader(Player):
     """A player that looks ahead at what the opponent will do and decides what to do."""
 
     name = 'Mind Reader'
-    stochastic = True # Don't cache me
+    behaviour = {
+        'memory_depth': -10,
+        'stochastic': False,
```

Comment:
> I think this one is stochastic, since the simulated play could be stochastic. MindReader alters opponent state rather than the source (so those two booleans are flipped).

I thought the same, __but__ it's only actually stochastic if it plays against a stochastic player. MindReader itself is not stochastic so should be cached against another non stochastic player.